### DISCUSSION TITLE
Automatically set `artist_sort` similar to the `the` plugin?

### DISCUSSION THREAD

**Discussion by frostyfalls** [beetbox/beets#5377] [Q&A]
Hi all,
I'm looking for a way to automatically set the `artist_sort` field similar to how the `the` plugin works. My goal is to have it set the `artist_sort` to what's in the `artist` for all tracks, but run a check similar to how the `the` plugin does for path templating.

I figure I can achieve this using some regex and the [importreplace](https://github.com/edgars-supe/beets-importreplace) plugin, but I wanted to see if anyone had a better idea.

Thanks!

**Comment by frostyfalls** [comment]
After a bit of playing around, I realize I can use path templates with `beet modify`, which works exactly how I want. Now I only wonder how I can have this behavior happen upon import. Is this what the `hook` plugin is for?

**Reply by frostyfalls** [reply]
I've almost got it. I'm trying to use `import.set_fields`, and for some reason `$artist` and `$artists` never gets parsed as a tag.

```yaml
import:
  set_fields:
    artist_sort: %the{$artist}
```

The example above just returns "$artist" for the `artist_sort`. The `%the` function gets executed properly, because I can do `%the{$albumartist}`, etc. `$artist` is just the standout here.

Even weirder, `$artists` just returns this:

```
artists_sort: $\␀a\␀r\␀t\␀i\␀s\␀t\␀s
```

**Reply by twebr** [reply]
I am facing the same issue as you do. As you already mention, your proposed solution _does_ work for the `albumartist` field:

```yaml
import:
  set_fields:
    albumartist_sort: '%the{$albumartist}'
```

This leads me to the working theory that `set_fields` can only access **album-level** fields (like `albumartist`), not **track-level** fields (like `artist`).

I am not sure if this brings us closer to any solution, but I wanted to share my findings with you.

**Reply by twebr** [reply]
I have just tried to import an album with the `-s` flag to enable singleton mode. And indeed: now the fields `artist` (and `artists`, for that matter) _are_ written correctly to the database. I guess this is a possible workaround, **but** it may lead to other problems along the way. Examples: you lose guarantees that album-level metadata is consistent within-album, and using `beet move` will put the tracks in your designated `singleton` location. Probably things that you do not want.

Any help from others in the community much appreciated.
